File: happyPanda.py
def happyPanda(coins, candies):
    if len(coins) == 0:
        return 0
    MAX = 0
    remainingCoin = 0
    ans = 0

    for i in range(len(coins)):
        remainingCoin += coins[i]

        price = candies[i][0]
        happiness = candies[i][1]

        if remainingCoin >= price:
            ans += happiness
            remainingCoin -= price
            MAX = max(MAX, happiness)
        else:
            if happiness >= MAX:
                ans -= happiness
    return ans

File: happyPanda.js
function happyPanda(coins, candies) {
    if (coins.length == 0) return 0;

    let max = 0;
    let ans = 0;
    let remainingCoin = 0;

    for (let i = 0; i < coins.length; i++) {
        remainingCoin += coins[i];
        let price = candies[i][0];
        let happiness = candies[i][1];

        if (remainingCoin >= price) {
            ans += happiness;
            remainingCoin -= price;
            max = Math.max(max, happiness);
        } else {
            if (happiness >= max) {
                ans -= happiness;
            }
        }
    }

    return ans
}